### PR TITLE
Add velero node-agent for Cluster Backup integration

### PR DESCRIPTION
**Pull request by moelsayed**
**What this PR does / why we need it**:
This PR adds the Velero node-agent daemonset to the user-cluster deployed resources. Node-agent is required to enable Persistent volume file system backups.
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature
**Special notes for your reviewer**:
The PR moves some resources that used to be deployed on the seed cluster side to the user cluster for the sake of simplification.
**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
- Deploy all Velero components on user cluster when backup is enabled.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: pkg/ee/cluster-backup/resources/user-cluster/cluster_backup.go [hunk 1]
```diff
@@ -43,12 +43,28 @@ const (
...
 		return resources.ClusterBackupNamespaceName, func(ns *corev1.Namespace) (*corev1.Namespace, error) {
+			ns.Labels = map[string]string{
+				"kubernetes.io/metadata.name": "velero",
+
+				"pod-security.kubernetes.io/audit":           "privileged",
+				"pod-security.kubernetes.io/audit-version":   "latest",
+				"pod-security.kubernetes.io/enforce":         "privileged",
+				"pod-security.kubernetes.io/enforce-version": "latest",
+				"pod-security.kubernetes.io/warn":            "privileged",
+				"pod-security.kubernetes.io/warn-version":    "latest",
```
Comment (embik) [inline pkg/ee/cluster-backup/resources/user-cluster/cluster_backup.go:67]: Why is this added here?

**Review by embik — COMMENTED** [review]
This PR does more than the description says, it also moves existing components into the user cluster. Why is this changed? Seems like an architectural change.

File: pkg/ee/cluster-backup/resources/user-cluster/cluster_backup.go (hunk 1 above)
Comment (moelsayed) [reply pkg/ee/cluster-backup/resources/user-cluster/cluster_backup.go:67]: The labels are generated by the `velero install` command. I added them for compatibility with upstream configuration.

Comment (embik) [reply pkg/ee/cluster-backup/resources/user-cluster/cluster_backup.go:67]: Okay, if this is from generated commands, can this be documented as a comment so people know how to generate this information for newer velero versions?

Comment (moelsayed) [reply pkg/ee/cluster-backup/resources/user-cluster/cluster_backup.go:67]: done. Good catch!

**Review by embik — APPROVED** [review]
/approve
